x-pack/plugins/security_solution/public/common/components/exceptions/use_fetch_or_create_rule_exception_list.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { HttpStart } from '../../../../../../../src/core/public';
+
+import {
+  ExceptionListSchema,
+  CreateExceptionListSchema,
+} from '../../../../../lists/common/schemas';
+import { Rule } from '../../../alerts/containers/detection_engine/rules/types';
+import { List, ListArray } from '../../../../common/detection_engine/schemas/types';
+import { fetchRuleById, patchRule } from '../../../alerts/containers/detection_engine/rules/api';
+import { fetchExceptionListById, addExceptionList } from '../../../lists_plugin_deps';
+
+export type ReturnUseFetchOrCreateRuleExceptionList = [boolean, ExceptionListSchema | null];
+
+export interface UseFetchOrCreateRuleExceptionListProps {
+  http: HttpStart;
+  ruleId: Rule['id'];
+  exceptionListType: ExceptionListSchema['type'];
+  onError: (arg: Error) => void;
+  onSuccess?: (arg: ExceptionListSchema) => void;
+}
+
+/**
+ * Hook for fetching or creating an exception list
+ *
+ * @param http Kibana http service
+ * @param ruleId id of the rule
+ * @param exceptionListType type of the exception list to be fetched or created
+ * @param onError error callback
+ * @param onSuccess optional callback when all lists fetched successfully
+ *
+ */
+export const useFetchOrCreateRuleExceptionList = ({
+  http,
+  ruleId,
+  exceptionListType,
+  onError,
+  onSuccess,
```

Comment:
Do you need the `onSuccess` or can that be inferred from the return value?

Reply:
Don't need it. Removed.